### ISSUE TITLE
Craft.findDeltaData has a significant impact on Entry Edit CP Front-End Performance.

### ISSUE THREAD

**Issue by internetztube** [craftcms/cms#9575]
### Description
I have a large content builder with up to 8 nested levels. 
```
Matrix -> SuperTable -> Matrix -> SuperTable -> ...
```

With Craft 3.7 I noticed that the browser stopped responding for about 2-3 seconds in the process of saving a delta draft. This is particularly annoying because the content editor cannot make any further changes in the control panel during this time. 
I dug through the code a bit and found out that these lines put a special load on the main thread. 
https://github.com/craftcms/cms/blob/develop/src/web/assets/cp/src/js/DraftEditor.js#L661-L665

This can also be seen very well in this Chrome Performance Report.
![Screenshot 2021-07-15 at 10 18 24](https://user-images.githubusercontent.com/6382364/125754509-119cb5d2-0df7-46c9-8c71-9f40b9e3c9b9.png)

This front-end performance issue is gone if you overwrite `Craft.findDeltaData` with a dummy function. Yes, of course the whole entry is now being saved. But the content editor can still make changes in the CP. 

```js
Craft.findDeltaData = (oldData, newData, deltaNames) => newData
```

Edit:
It looks like this commit is causing the issue.
https://github.com/craftcms/cms/commit/c474a8a3e24b210b8a531f0e112132a698484dc6#

### Additional info

- Craft version: 3.7.0
- PHP version: 7.3.28
- Database driver & version: PostgreSQL 10.6

**Comment by brandonkelly** [comment]
Can you please send a database backup and your `composer.json` + `composer.lock` files into support@craftcms.com, and point us to an entry where we can test? Will speed up our ability to test possible performance fixes.

**Comment by internetztube** [comment]
Sent! @brandonkelly

**Comment by internetztube** [comment]
Also sent over the other requested files. @brandonkelly

**Comment by brandonkelly** [comment]
Thanks! Found a way to revert the offending change in c474a8a3e24b210b8a531f0e112132a698484dc6 without reintroducing the bug it fixed. There’s still a bit of a lag, but it’s on par with 3.6 now. Further improvements are going to require a larger refactoring effort, which will need to happen in v4.

**Comment by internetztube** [comment]
It looks like the bugfix spawned another bug. 😞
When you apply a change to a field and then undo it, the change is not reflected. I sent a video about this to our Zendesk Conversation.

**Comment by brandonkelly** [comment]
@internetztube Doh, thanks for pointing that out. Released 3.7.3.2 with a fix.

**Comment by internetztube** [comment]
I found the root cause of the poor front-end performance. Please have a look! @brandonkelly 
https://github.com/craftcms/cms/pull/9605
